### DISCUSSION TITLE
Cancel deferred action?

### DISCUSSION THREAD

**Discussion by modernben** [livewire/livewire#4102] [1. Help]
I currently have a list of radio buttons wired via `wire:model.defer="myfield.{{ some-id }}.play"`

Client wants a "clear" button to set the values back to unchecked.
Thought a button type="reset" would do it but the action is still in the deferredActions list and being fired on the next event.

How do I cancel a deferred action?

**Comment by modernben** [comment]
FWIW, currently doing this and it seems to be working.

```javascript
$('.reset-play').click(function(){
    var component = @this.__instance;

    for(property in component.deferredActions) {
        delete component.deferredActions[property];
    }
});
```

**Comment by joshhanley** [comment]
@modernben I probably would just have a button that calls an action (maybe using `$set`) that resets the value back to its default rather than use the javascript you have above.

It will mean that the deferred action is still there, but your component will just receive two actions, one with the original set and the next with the reset, which is fine in my opinion.

The reason you shouldn't use your above javascript code, is you are doing something that's not supported, so if we change the internals, your app would break.

Hope this helps!
